apps/functions-subscription/src/adapters/azure/functions/__tests__/subscriptions.test.ts

```diff
@@ -1,78 +1,129 @@
 import { describe, expect, it } from 'vitest';
 import * as TE from 'fp-ts/TaskEither';
 import { HttpRequest } from '@azure/functions';
-import { makeAValidCreateSubscriptionRequest } from './data';
+import {
+  makeAValidCreateSubscriptionRequest,
+  makeAValidGetSubscriptionRequest,
+} from './data';
 import { makeFunctionContext, makeTestSystemEnv } from './mocks';
-import { makePostSubscriptionHandler } from '../subscriptions';
+import {
+  makeGetSubscriptionHandler,
+  makePostSubscriptionHandler,
+} from '../subscriptions';
 import { SubscriptionStoreError } from '../../../../use-cases/errors';
 import { aSubscription } from '../../../../domain/__tests__/data';
-import { ItemAlreadyExists } from '../../../../domain/errors';
+import { ItemAlreadyExists, ItemNotFound } from '../../../../domain/errors';
 
 describe('subscriptions azure function', () => {
-  it('should return 201 with the created subscription', async () => {
-    const env = makeTestSystemEnv();
-    env.insertSubscription.mockReturnValueOnce(TE.right(aSubscription));
-
-    const actual = await makePostSubscriptionHandler(env)(
-      makeAValidCreateSubscriptionRequest(),
-      makeFunctionContext(),
-    );
-    expect(actual.status).toStrictEqual(201);
-  });
+  describe('insert subscription', () => {
```

Comment:
**note**: lot of changes here, but just because I wrapped already present unit tests in a dedicated `describe`.